src/mjolnir/landmark_builder.cc

```diff
@@ -16,12 +66,20 @@ struct LandmarkDatabase::db_pimpl {
   bool vacuum_analyze = false;
 
   db_pimpl(const std::string& db_name, bool read_only) : insert_stmt(nullptr) {
-    // figure out if we need to create it or can just open it up
-    auto flags = read_only ? SQLITE_OPEN_READONLY : SQLITE_OPEN_READWRITE;
+    // create parent directory if not exist
+    const filesystem::path parent_dir = filesystem::path(db_name).parent_path();
+    if (!filesystem::exists(parent_dir) && !filesystem::create_directories(parent_dir)) {
+      throw std::runtime_error("Can't create parent directory " + parent_dir.string());
+    }
+
+    // figure out if we need to create database or can just open it up
+    auto flags = read_only ? SQLITE_OPEN_READONLY : SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE;
     if (!filesystem::exists(db_name)) {
       if (read_only)
         throw std::logic_error("Cannot open sqlite database in read-only mode if it does not exist");
-      flags |= SQLITE_OPEN_CREATE;
+    } else if (!read_only) {
+      std::filesystem::remove(db_name);
```

Comment:
long story but we only just went to c++17 which as non-experimental support for filesystem. before that we use valhalla::filesystem, our internal home grown version. we've not migrated to std::filesystem yet so lets use our own until we do so

```suggestion
      filesystem::remove(db_name);
```